flower/views/tasks.py

```diff
@@ -83,6 +83,40 @@ def key(item):
                         recordsTotal=len(sorted_tasks),
                         recordsFiltered=len(sorted_tasks)))
 
+    @web.authenticated
+    def post(self):
+        app = self.application
+        draw = self.get_argument('draw', type=int)
+        start = self.get_argument('start', type=int)
+        length = self.get_argument('length', type=int)
+        search = self.get_argument('search[value]', type=str)
+
+        column = self.get_argument('order[0][column]', type=int)
+        sort_by = self.get_argument('columns[%s][data]' % column, type=str)
+        sort_order = self.get_argument('order[0][dir]', type=str) == 'desc'
+
+        def key(item):
+            return Comparable(getattr(item[1], sort_by))
+
+        sorted_tasks = sorted(
+            iter_tasks(app.events, search=search),
+            key=key,
+            reverse=sort_order
+        )
+
+        filtered_tasks = []
+
+        for task in sorted_tasks[start:start + length]:
+            task_dict = as_dict(self.format_task(task)[1])
+            if task_dict.get('worker'):
+                task_dict['worker'] = task_dict['worker'].hostname
+
+            filtered_tasks.append(task_dict)
+
+        self.write(dict(draw=draw, data=filtered_tasks,
+                        recordsTotal=len(sorted_tasks),
+                        recordsFiltered=len(sorted_tasks)))
+
```

Comment:
Looks like `get` and `post` functions have identical body. It would be better to define `post` function with `post = get`.

Reply:
Will do. thanks.

Reply:
Done.

Reply:
Thanks @johnarnold . I will merge this PR after fixing the tests.